newsroom/agenda/utils.py

```diff
@@ -185,3 +187,97 @@ def get_item_type(item: Dict[str, Any]) -> Literal["event", "planning"]:
         return "event"
     else:
         return "planning"
+
+
+def coverage_is_completed(coverage: Dict[str, Any]) -> bool:
+    return coverage.get("workflow_status") == ASSIGNMENT_WORKFLOW_STATE.COMPLETED
+
+
+def remove_restricted_coverage_info(items):
+    keys_to_copy = (
+        "coverage_id",
+        "coverage_type",
+        "planning_id",
+        "watches",
+        "planning",
+        "delivery_id",
+        "delivery_href",
+        "deliveries",
+    )
+
+    for item in items:
+        item["coverages"] = [
+            coverage
+            if coverage_is_completed(coverage)
+            else {key: val for key, val in coverage.items() if key in keys_to_copy}
+            for coverage in item.get("coverages") or []
+        ]
+
+        for plan in item.get("planning_items") or []:
+            plan["coverages"] = [
+                coverage
+                if coverage_is_completed(coverage)
+                else {key: val for key, val in coverage.items() if key in keys_to_copy}
+                for coverage in plan.get("coverages") or []
+            ]
+            for coverage in plan["coverages"]:
+                if not coverage_is_completed(coverage):
+                    coverage["planning"] = {"g2_content_type": coverage["planning"]["g2_content_type"]}
+
+    return items
+
+
+def get_restricted_companies_tuple() -> Tuple[Set[str], Set[str]]:
+    restricted_companies = set()
+    unrestricted_companies = set()
+    for company in query_resource("companies"):
+        if company.get("restrict_coverage_info"):
+            restricted_companies.add(company.get("_id"))
+        else:
+            unrestricted_companies.add(company.get("_id"))
+
+    # Make sure to send notifications for admins without a company
+    unrestricted_companies.add("None")
+
+    return restricted_companies, unrestricted_companies
+
+
+def push_agenda_items_notification(name, items, **kwargs):
+    restricted_companies, unrestricted_companies = get_restricted_companies_tuple()
+
+    if not len(restricted_companies):
+        push_notification(name, _items=items, **kwargs)
+    else:
+        push_notification(
+            name,
+            {"exclude": {"company": list(restricted_companies)}},
+            _items=items,
+            **kwargs,
+        )
+        push_notification(
+            name,
+            filters={"include": {"company": list(restricted_companies)}},
+            _items=remove_restricted_coverage_info(items),
+            **kwargs,
+        )
+
+
+def push_agenda_item_notification(name, item, **kwargs):
```

Comment:
not sure if we need both methods, in theory this could be handled by calling:
```
push_agenda_items_notification("name", [item])
```
which would avoid some duplication